OpenRA.Mods.Common/Warheads/DamageWarhead.cs

```diff
@@ -38,9 +38,19 @@ public override bool IsValidAgainst(Actor victim, Actor firedBy)
 
 		public int DamageVersus(Actor victim)
 		{
-			var armor = victim.TraitsImplementing<Armor>()
-				.Where(a => !a.IsTraitDisabled && a.Info.Type != null && Versus.ContainsKey(a.Info.Type))
-				.Select(a => Versus[a.Info.Type]);
+			// If no Versus values are defined, DamageVersus would return 100 anyway, so we might as well do that early.
+			if (Versus.Count == 0)
+				return 100;
+
+			// PERF: Avoid LINQ
+			var armor = Enumerable.Empty<int>();
+			foreach (var a in victim.TraitsImplementing<Armor>())
+			{
+				if (a.IsTraitDisabled || a.Info.Type == null || !Versus.ContainsKey(a.Info.Type))
+					continue;
+
+				armor.Append(Versus[a.Info.Type]);
```

Comment:
Because of the `Enumerable.Empty` above, this is actually handled by LINQ in a much less efficient way than we have above!

You make `armor` a list, but you still need to enumerate the collection twice (once to build the list, and then again inside `ApplyPercentageModifiers`), so better to drop this part of the commit.